### PR TITLE
[HomeWizard] Add support for Energy Socket and Watermeter products

### PR DESCRIPTION
**Pull request by Daniel-42**
The HomeWizard productline has a few new devices. Among them a Watermeter and an Energy Socket.

In this change I have added support for these by adding two new things to the binding: the Watermeter and the Energy Socket.

https://openhab.jfrog.io/ui/native/libs-pullrequest-local/org/openhab/addons/bundles/org.openhab.binding.homewizard/

Signed-off-by: Daniël van Os <daniel@supercell.nl>

### CODE REVIEW

**Comment by lsiepel** [comment]
@Daniel-42 Unfortunately this PR has been here for a while without a review. I could do a basic review, but before i do it would be good to fix the conflicts and bring the branch up to date. 
Are the changes in this PR tested?

**Comment by bertjan** [comment]
I'm happy to test this. I tried adding the JAR file referenced above to my addons folder and although that activates the P1 meter thing I don't see the water meter capabilities in the addon.

**Comment by lsiepel** [comment]
@Daniel-42 if you proceed with this PR, it will also need thing upgrade instructions: https://www.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types

**Comment by ababilone** [comment]
Hi, any news on this PR ? I'm interested in it too, do you need help on anything ? Thanks!

**Comment by lsiepel** [comment]
This PR was hanging around for almost 2 years without any progress, it also looks pretty good. So i took the oppertunity to try to bring it to the finish line. As i changed some minor things, another @openhab/add-ons-maintainers needs to review and merge. 

Before that it would be nice to have some test results from @ababilone or @bertjan or whoever can test it. My changes are limited to:
- Fixed the merge conflict
- Fixed the license headers
- Adjust the channel labels + changing brightness from dimmer to number:dimensionless
- Adjust the readme 

Test jar (openHAB 4.2.0+) https://1drv.ms/u/s!AnMcxmvEeupwjvhXkfIKBYTP1txbyg?e=hxhx5D

File: bundles/org.openhab.binding.homewizard/src/main/resources/OH-INF/thing/thing-types.xml
Comment (wborn) [inline bundles/org.openhab.binding.homewizard/src/main/resources/OH-INF/thing/thing-types.xml:1]: Update the default translations properties file so the new strings can be translated.

**Review by wborn — COMMENTED** [review]
I see a few minor improvements. :slightly_smiling_face:

File: bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/HomeWizardStatefulDeviceHandler.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    /**
+     * Device specific handling of the returned state payload.
+     *
+     * @param payload The data parsed from the state Json file
+     */
+    abstract protected void handleStatePayload(StatePayload payload);
+
+    /**
+     *
+     */
```
[the remark this answers is not in the document]
Comment (lsiepel) [reply bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/HomeWizardStatefulDeviceHandler.java:54]: Conflict left over, thanks.

**Review by wborn — APPROVED** [review]
Thanks!

**Pull request merged**